src/store/modules/models/mutations.js

```diff
@@ -38,37 +38,72 @@ export default {
     );
   },
 
-    updateStoryWithData (state, payload) {
-        const story = payload.story;
-        delete payload.story;
-        Object.assign(story, payload);
-    },
-    createImageForStory (state, payload) {
-        const story = state.stories.find(s => s.id === payload.story_id);
-        story.images.push(payload.image);
-    },
+  updateStoryWithData(state, payload) {
+    const story = payload.story;
+    delete payload.story;
+    Object.assign(story, payload);
+  },
+  createImageForStory(state, payload) {
+    const story = state.stories.find(s => s.id === payload.story_id);
+    story.images.push(payload.image);
+  },
+
+  /**
+   * Updates the basic data stored on the space
+   * If the building type is changed, it will affect all other spaces in the same building unit
+   * If the building unit is changed, it will change the type of the building to match all other spaces with that building unit
+   *
+   * @param {*} state State to update
+   * @param {*} payload Payload with data fields to update on the space
+   */
+  updateSpaceWithData(state, payload) {
+    const space = payload.space;
+
+    if (payload.building_type_id && space.building_unit_id) {
+      state.stories.forEach((story) => {
```

Comment:
I'm not sure if a map would be more performant here per the comment you left on my last PR. https://github.com/NREL/floorspace.js/pull/399#discussion_r673392804